### ISSUE TITLE
[Bug]: Frequent Crashes Using Mekanism Teleporters (AIR_LEAK sound)

### ISSUE THREAD

**Issue by zarillion** [FTBTeam/FTB-Modpack-Issues#1071]
### Modpack

FTB Plexiglass Mountain

### Modpack version

1.1.1

### Log Files

[crash-2022-06-28_21.42.51-client.txt](https://github.com/FTBTeam/FTB-Modpack-Issues/files/9050269/crash-2022-06-28_21.42.51-client.txt)
[crash-2022-07-05_11.45.41-client.txt](https://github.com/FTBTeam/FTB-Modpack-Issues/files/9050270/crash-2022-07-05_11.45.41-client.txt)
[crash-2022-07-05_12.16.21-client.txt](https://github.com/FTBTeam/FTB-Modpack-Issues/files/9050271/crash-2022-07-05_12.16.21-client.txt)
[crash-2022-07-05_13.35.56-client.txt](https://github.com/FTBTeam/FTB-Modpack-Issues/files/9050272/crash-2022-07-05_13.35.56-client.txt)
[crash-2022-07-05_18.39.53-client.txt](https://github.com/FTBTeam/FTB-Modpack-Issues/files/9050273/crash-2022-07-05_18.39.53-client.txt)

### Describe the bug

I created a floating platform of portals in the AE2 Spatial IO dimension to serve as our hub. At the same time my friend was progressing in PneumaticCraft and placed a few Safety Tube Modules which frequently play the AIR_LEAK sound. Using these portals often leads to a client-side crash:

```
java.lang.IllegalArgumentException: Invalid moving sound AIR_LEAK for focus object BlockPos{x=154, y=79, z=204}
```

The coordinates in each crash report point to a PneumaticCraft Safety Tube Module, which plays the AIR_LEAK sound every few seconds.

### Steps to reproduce

1. Create a Mekanism Teleporter (unclear if it needs to be in/out of Spatial IO or not).
2. Attach a Safety Tube Module to a pressure tube. We are using a Flux Compressor w/ 4 speed upgrades to pressurize the tube.
3. Use the teleporter until the client crashes.

### Expected behaviour

No client crashes while using our teleporters

### Screenshots

![image](https://user-images.githubusercontent.com/64381982/177435862-27898074-2b5d-4aaa-b7f7-0ae3e148cbcd.png)
![image](https://user-images.githubusercontent.com/64381982/177435903-ec064f06-c144-4882-bd49-3b9fbd219979.png)

**Comment by zarillion** [comment]
Reading the code in `MovingSounds.java` for PNC, my guess is that this *does* have something to do with the portals crossing into different dimensions:

https://github.com/TeamPneumatic/pnc-repressurized/blob/1.18.2/src/main/java/me/desht/pneumaticcraft/client/sound/MovingSounds.java#L84-L89

The `BlockEntity te = world.getBlockEntity(pos);` assignment may be returning `null` because it is checking for the Safety Tube Module in the wrong dimension as the player teleports. It decides there is nothing there to play a sound for, breaks out of the `switch` and throws the exception. It seems like this could happen any time a player enters a new dimension and the sound is trying to play.

**Comment by Raidobw2** [comment]
Thanks for the report and insight, I've relayed that to the author above ^

Also, that is a nice portal hub! It looks good :)

**Comment by Raidobw2** [comment]
The mod author released a patch at https://github.com/TeamPneumatic/pnc-repressurized/actions/runs/2634619301 for it. If you have time you can check it out, or I'll simply include the fixed build (when released on CF) on the next PGM update

**Comment by zarillion** [comment]
Thanks for the quick turn-around on this! I'll play around with the development `.jar` and look forward to the update.

Also adding another shameless request for CC: Tweaked!

**Comment by Raidobw2** [comment]
I've had many requests for CC. I'm planning to add it, yes :)

**Comment by Raidobw2** [comment]
Fixed in 1.2.0
And added CC! Plus two CC addons
